modelling_earth/velocity.py

```diff
@@ -0,0 +1,84 @@
+"""
+Functions to create 2D and 3D velocity distributions
+"""
+
+import numpy as np
+import xarray as xr
+
+
+def linear_velocity(coordinates, z_start, velocity_bottom, direction="x"):
+    """
+    Hotizontal velocity distribution along a define direction.
+    The velocity will be zero above the z_start and will linearly increase bellow that
+    depth until the bottom of the grid. The velocity distribution will be defined on
+    both lateral boundaries.
+
+    Parameters
+    ----------
+    coordinates : :class:`xarray.DataArrayCoordinates`
+        Coordinates located on a regular grid where the temperature distribution will be
+        created. Must be in meters and can be either 2D or 3D.
+    z_start : float
+        Depth value where the velocity condition changes in meters.
+    velocity_bottom : float
+        Velocity at bottom boundary (z_min) on the x and z axes in m/s.
+    direction : string (optional)
+        Direction of the subduction. If working in 3D it can be either *"x"* or *"y"*.
+        When working in 2D, it must be *"x"*.
+
+    Returns
+    -------
+    velocity : :class:`xarray.Dataset`
+        Dataset containing the velocity distribution.
+    """
+    # Check if coordinates is 2D or 3D and if velocity dims are corrected
+    dimension, expected_dims = _check_velocity(coordinates, velocity_bottom)
+    # Define the shape of the coordinates
+    shape = tuple(coordinates[i].size for i in expected_dims)
+    # Initialize the velocity dataset with zero values
+    velocity = xr.Dataset(coords=coordinates)
+    for i in expected_dims:
+        velocity["velocity_{}".format(i)] = (coordinates, np.zeros(shape))
+
+    # Calculate the velocity.
+    # Assume a linear increase with depth where where the velocity is 0 for
+    # coordines.z > z_start and for coordinates.z[0] < z < z_start.
+    # These velocity values must be added on the first and last column.
+    # Define the condition to calculate the velocity
+    condition = np.logical_and(
+        velocity.z < z_start,
+        np.logical_or(velocity.x == velocity.x[0], velocity.x == velocity.x[-1]),
```

Comment:
You forgot to change `x` with the passed `direction`. Remember you can use `getattr` to get an attribute of a class by a string variable. So for example, in this case you can use `getattr(velocity, direction)` to get the desired coordinate.

Reply:
Done!

Reply:
Done! Thank you for the data